### PR TITLE
Fulfill auth path tests logic

### PR DESCRIPTION
**Pull request by c633**
We should check the index for proof of inclusion/absence, instead of `leaf.Value() == nil`

### CODE REVIEW

**Comment by arlolra** [comment]
LGTM

File: merkletree/proof_test.go [hunk 1]
```diff
@@ -53,6 +53,13 @@ func TestVerifyProof(t *testing.T) {
 
 	// proof of inclusion
 	proof := m.Get(index3)
+	if proof.Leaf.Value() == nil {
+		t.Fatal("Expect returned leaf's value is not nil")
+	}
+	// ensure this is a proof of inclusion
+	if !bytes.Equal(proof.LookupIndex, proof.Leaf.Index()) {
```
Comment (masomel) [inline merkletree/proof_test.go]: I think I would feel more comfortable checking that ``proof.LookupIndex == privateIndex(name) && LookupIndex == proof.Leaf.Index()``. It only becomes a proof of inclusion once both of these conditions hold.

Comment (c633) [inline merkletree/proof_test.go]: Do you mean this check `proof.LookupIndex == privateIndex(name)` is  
https://github.com/coniks-sys/coniks-go/blob/c04f1e2d88fe5edc35add235942a7b95162d8fe9/merkletree/proof_test.go#L64 ?

Comment (masomel) [inline merkletree/proof_test.go]: Yup. I know the tests are already checking both conditions individually. I know this is nitpicky, but what I'm trying to say is that, really specifically, ``proof.LookupIndex == proof.Leaf.Index()`` alone doesn't imply it's a proof of inclusion for the name we're interested in.

Comment (c633) [inline merkletree/proof_test.go]: Yeah, I got it. Nice!

Comment (c633) [inline merkletree/proof_test.go]: Fixed it in d8d62300be40538c5e487c5e6f308cc4c8161c18.

File: merkletree/proof_test.go [hunk 2]
```diff
@@ -53,9 +53,14 @@ func TestVerifyProof(t *testing.T) {
...
 	proof := m.Get(index3)
-	// step 1. verify VRF index
-	if !bytes.Equal(vrfPrivKey1.Compute([]byte(key3)), proof.LookupIndex) {
-		t.Error("VRF verification returns false")
+	if proof.Leaf.Value() == nil {
+		t.Fatal("Expect returned leaf's value is not nil")
+	}
+	// step 1. ensure this is a proof of inclusion by comparing the returned indices
+	// and verifying the VRF index as well.
+	if !bytes.Equal(proof.LookupIndex, proof.Leaf.Index()) ||
```
Comment (arlolra) [inline merkletree/proof_test.go]: you can do the same consolidation for the absence as well

Comment (c633) [inline merkletree/proof_test.go]: Done in 2802113177ddfee77630bdb930f4a2c0629b95d5.

Comment (arlolra) [inline merkletree/proof_test.go]: Yeah, good stuff.

**Comment by arlolra** [comment]
@masomel merge?

**Comment by masomel** [comment]
Yeah, LGTM

**Pull request closed without merging**